LearningJourney/Modules/Features/JLibrary/JLibraryTests/Sources/Scenes/ObjectivesLisst/Domain/ObjectivesListViewModelTests.swift

```diff
@@ -0,0 +1,152 @@
+//
+//  ObjectivesListViewModelTests.swift
+//  JLibraryTests
+//
+//  Created by Maria Fernanda Azolin on 08/10/21.
+//
+
+import TestingUtils
+import XCTest
+
+@testable import JLibrary
+
+final class ObjectivesListViewModelTests: XCTestCase {
+    
+    private let fetchObjectivesUseCaseSpy = FetchObjectivesUseCaseSpy()
+    private let toggleLearnUseCaseSpy = ToggleLearnUseCaseSpy()
+    private let goalMock: LearningGoal = .init(id: 1, name: "goal", progress: 0.0)
+    
+    private lazy var sut = ObjectivesListViewModel(
+        useCases: .init(
+            fetchObjectivesUseCase: fetchObjectivesUseCaseSpy,
+            toggleLearnUseCase: toggleLearnUseCaseSpy
+        ),
+        dependencies: .init(goal: goalMock)
+    )
+    
+    // MARK: - handleOnAppear
+    
+    func test_handleOnAppear_whenResultIsSuccess_shouldCorrectlyLoadObjectives() {
+        //Given
+        fetchObjectivesUseCaseSpy.objectiveToUse = .success([])
+        
+        //When
+        sut.handleOnAppear()
+        
+        //Then
+        XCTAssertEqual(sut.objectives, .result([]))
+        
+        let learningGoal = LearningGoal.init(id: 1, name: "goal", progress: 0)
```

Comment:
Consider renaming this to better match it's role in the test
Also, create a fixture :)
```suggestion
        let expectedLearningGoal = LearningGoal.init(id: 1, name: "goal", progress: 0)
```